StevesSixLeggedSprinters/BangazonAPI
# Km Department

## Status
Choose:
**READY**

## Migrations
YES 

## Description
I created a Department model (Dept.cs) in order to create a table in the database for all departments. I also created a department controller (DeptController.cs) so we are able to GET a list of departments, GET a single department, POST (create) a department, and PUT (edit) a specific department. Also modified BangazonAPIContext file to set up a relationship between the database and the department table. 


## Todos
- [ Y] Tests
- [ Y] Documentation


## Deploy Notes
git fetch --a  (fetch all branches from GitHub)
git branch -avv (display all branches)
git pull origin km-dept (pull my code down to test)
dotnet ef migrations add Initial (migrations need to be built)
dotnet ef database update (update the database since a new migration has been built)
open bangazon.db (open database)
dotnet run

## Steps to Test or Reproduce
I have set up the following methods: GET, POST, PUT
Follow these step by step in order to test.

git fetch --a  (fetch all branches from GitHub)
git branch -avv (display all branches)
git checkout km-dept (switch into my branch)
dotnet ef migrations add Initial (migrations need to be built)
dotnet ef database update (update the database since a new migration has been built)
open bangazon.db (open database)
dotnet run

Make sure to have Postman installed: https://www.getpostman.com/
In Postman, 

To add a department:
Select POST
http://localhost:5000/api/DeptType

And add this to the body before sending.
Example:
{		
"DeptTypeId":"1",
"DeptName":"IT",
"ExpBudget":"3mil"
}

To view all departments:
Select GET
http://localhost:5000/api/DeptType

To view a single department:
Select GET
http://localhost:5000/api/DeptType/{id}

To edit a single department
Select PUT
http://localhost:5000/api/DeptType/{id}

## Impacted Areas in Application
List general components of the application that this PR will affect:
Employees will use the DeptTypeId as a foreign key.
* 
https://github.com/StevesSixLeggedSprinters/BangazonAPI/issues/7
